src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -190,6 +212,29 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     } flatMap {
       case Some(existingCluster) => throw ClusterAlreadyExistsException(googleProject, clusterName, existingCluster.status)
       case None =>
+        clusterRequest.enableWelder match {
```

Comment:
This match statement doesn't look correct -- it returns a Future but that Future isn't actually chained to anything, so it probably doesn't have any effect.

Reply:
I'm also hazy on why we're checking other clusters in the project here. Did that requirement come from @sharmatime? Why should welder enablement be dependent on other clusters in the project?

Reply:
I think we're planning to remove this enableWelder field once welder has been released to both AoU and Terra. For the time between those releases, this is just a check that both welder-enabled and welder-disabled clusters don't exist in the same google project, since it would mean that people's work could get clobbered even though welder was on the cluster

Reply:
I'm not really a fan of `createCluster` querying _other_ users' clusters, which the calling user may not have permission to do. I feel this logic lies outside of the responsibility of Leo. A project owner should be able to call `listClusters` to check welder enablement across clusters. The logic to roll out welder to projects should be handled by the parent app (e.g. AoU, Terra).

Reply:
Hm, I thought it would be similar to how we query for any other clusters in the same project with the same clusterName. Is the concern that the query returns a list of other users' clusters? I recently made a change so that the LeoService only sees a boolean that tells if there are welder enabled/disabled clusters in the project, rather than listing them, if that makes a difference. @Qi77Qi, was this check something that the apps asked for, or just part of the original design?